x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -180,18 +203,29 @@ export const useFetchIndex = (
 
       asyncSearch();
     },
-    [addError, data.dataViews, includeUnmapped, indexNames, state]
+    [
+      addError,
+      addWarning,
+      data.dataViews,
+      dataViewIdOrIndexNames,
+      includeUnmapped,
+      isIndexNameArray,
+      state,
+    ]
   );
 
   useEffect(() => {
-    if (!isEmpty(indexNames) && !isEqual(previousIndexesName.current, indexNames)) {
-      indexFieldsSearch(indexNames);
+    if (
+      !isEmpty(dataViewIdOrIndexNames) &&
+      !isEqual(previousIndexesNameOrId.current, dataViewIdOrIndexNames)
+    ) {
+      indexFieldsSearch(dataViewIdOrIndexNames);
```

Comment:
Oh yeah I thought I had it written this up somewhere but the plan is to refactor this to use react-query after this. I thought it would've been too much to change in this PR.